github/workflows/test.yml

```diff
@@ -10,6 +10,7 @@ on:
 jobs:
   lint_test:
     runs-on: ubuntu-latest
+    environment: test
```

Comment:
I thought that environments were for running infrastructure (i.e. CD, not CI). I see `RAILS_ENV: test` lower down, what exactly does this do?

Reply:
nfsa-pdc has environment variables like this:
```
          ACTIVE_RECORD_ENCRYPTION_PRIMARY_KEY:  ${{ secrets.ACTIVE_RECORD_ENCRYPTION_PRIMARY_KEY }}
```
I suppose it needs the "environment: test" to know which environment to get the secrets from?
since nfsa-pdc already has customisations to this workflow then maybe it's not worth adding this change just for nfsa-pdc

Reply:
I think that's a mistake in nfsa-pdc. That's not the right way to use secrets – they should be in an environment when they're part of a deployment script, not for a CI build.

Reply:
This has wider implications, e.g. when you set the environment then it changes the OIDC name for the action